bundle/statemgmt/state_load.go

```diff
@@ -0,0 +1,137 @@
+package statemgmt
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/deploy/terraform"
+	"github.com/databricks/cli/bundle/statemgmt/resourcestate"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+type (
+	ExportedResourcesMap = resourcestate.ExportedResourcesMap
+	ResourceState        = resourcestate.ResourceState
+	loadMode             int
+)
+
+const ErrorOnEmptyState loadMode = 0
+
+type load struct {
+	modes []loadMode
+}
+
+func (l *load) Name() string {
+	return "statemgmt.Load"
+}
+
+func (l *load) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
```

Comment:
This is still TF-specific. I expected the `StateToBundle` function to move here, but I still keep separate mutators for the different deployment modes. Doesn't that make more sense?